src/trials.jl

```diff
@@ -339,168 +355,194 @@ Base.show(io::IO, t::TrialRatio) = _show(io, t)
-    # if median size of (bins with >10% average data/bin) is less than 5% of max bin size, log the bin sizes
-    if logbins === true || (logbins === nothing && median(filter(b -> b > 0.1 * length(times) / histwidth, bins)) / maximum(bins) < 0.05)
-        bins, logbins = log.(1 .+ bins), true
-    else
-        logbins = false
+    logbins = get(io, :logbins, nothing) === true
+    caption = logbins ? ("log(counts) from " * samplesstr) : samplesstr
+
+    # The height and width of the printed histogram in characters:
+    histheight = 2    
+    _nonhistwidth = 5 + length(boxspace)
+    _minhistwidth = 18 + length(caption)
+    histwidth = max(_minhistwidth, min(90, displaysize(io)[2]) - _nonhistwidth)
+    # This should fit it within your terminal, but stops growing at 90 columns. Below about
+    # 55 columns it will stop shrinking, by which point the first line has already wrapped.
+
+    histmin = get(io, :histmin, hist_round_low(t.times, mintime, avgtime))
+    histmax = get(io, :histmax, hist_round_high(t.times, avgtime, quantime))
+
+    # Here nextfloat() ensures both endpoints included, will only matter for
+    # artificial cases such as:  Trial(Parameters(), [3,4,5], [0,0,0], 0, 0)
+    fences = range(histmin, nextfloat(float(histmax)), length=histwidth)
+    bins = histogram_bindata(t.times, fences)
+    # Last bin is everything right of last fence, introduce a gap for printing:
+    _lastbin = pop!(bins)
+    push!(bins, 0, _lastbin)
+    if logbins
+        bins = log.(1 .+ bins)
     end
     hist = asciihist(bins, histheight)
-    hist[:,end-1] .= ' '
-    maxbin = maximum(bins)
-
-    delta1 = (histmax - histmin) / (histwidth - 1)
-    if delta1 > 0
-        medpos = 1 + round(Int, (histtimes[length(times) ÷ 2] - histmin) / delta1)
-        avgpos = 1 + round(Int, (mean(times) - histmin) / delta1)
-    else
-        medpos, avgpos = 1, 1
+    hist[:, end-1] .= ' '
+
+    avgpos = searchsortedlast(fences, avgtime)
+    medpos = searchsortedlast(fences, medtime)
+    q25pos = searchsortedlast(fences, quantile(t.times, 0.25)) # might be 0, that's OK
+    q75pos = searchsortedlast(fences, quantile(t.times, 0.75))
+
+    # Above the histogram bars, print markers for special ones:
+    printstyled(io, pad, "│", boxspace; color=boxcolor)
+    istop = maximum(filter(i -> i in axes(hist,2), [avgpos, medpos+1, q75pos]))
+    for i in axes(hist, 2)
+        i > istop && break
+        if i == avgpos
+            printstyled(io, "*", color=avgcolor, bold=true)
+        elseif i == medpos || 
+                (medpos==avgpos && i==medpos-1 && medtime<=avgtime) ||
+                (medpos==avgpos && i==medpos+1 && medtime>avgtime)
+            # Marker for "median" is moved one to the left if they collide exactly
+            # printstyled(io, "½", color=medcolor)
+            printstyled(io, "◑", color=medcolor)
+        elseif i == q25pos
+            # Quartile markers exist partly to explain the median marker, without needing a legend
```

Comment:
Are the quartile markers too much clutter? I'm not sure they are very useful. 

The idea was to make the median marker self-explanatory. "quarter == quartile" isn't even a pun, right?

Reply:
After https://github.com/JuliaCI/BenchmarkTools.jl/pull/260/commits/ccb2bc34dfaa30ec9bff7042abc4f6f4246bf156 the quartile markers are now hidden, on Julia 1.7 and later. On 1.6 this option doesn't exist (or would require messing with terminal codes).

Maybe it's too weird to do two things, effectively. It does look less cluttered without them.